tests/unit/test_extras.hxx

```diff
@@ -255,16 +258,32 @@ public:
     output_info.disable();
     bout::globals::mesh->createDefaultRegions();
     output_info.enable();
+
+    // Delete any existing mesh_staggered
+    if (mesh_staggered != nullptr) {
+      delete mesh_staggered;
+      mesh_staggered = nullptr;
+    }
+    mesh_staggered = new FakeMesh(nx, ny, nz);
+    mesh_staggered->StaggerGrids = true;
+    output_info.disable();
+    static_cast<FakeMesh*>(mesh_staggered)->setCoordinates(nullptr, CELL_XLOW);
+    mesh_staggered->createDefaultRegions();
+    output_info.enable();
   }
 
   ~FakeMeshFixture() {
     delete bout::globals::mesh;
     bout::globals::mesh = nullptr;
+    delete mesh_staggered;
+    mesh_staggered = nullptr;
   }
 
   static constexpr int nx = 3;
   static constexpr int ny = 5;
   static constexpr int nz = 7;
+
+  Mesh* mesh_staggered = nullptr;
```

Comment:
I think I must be being a bit dim; why do we need to make a separate mesh for the tests involving staggering?

Reply:
It seemed neater to me. The tests with staggering were doing things like
```
  auto backup = mesh->StaggerGrids;
  mesh->StaggerGrids = true;
  ...
  mesh->StaggerGrids = backup;
```
and they also needed `mesh->setCoordinates(nullptr CELL_XLOW);` calls adding. By having `mesh_staggered` we only need to do the setup in one place.

If you prefer, I'd be happy to have staggering enabled in `mesh` for all the tests, but since `mesh->StaggerGrids=false` is the default, I assumed it would be better to have the tests using that unless they needed staggering.

Reply:
@ZedThree do you want any changes to the unit tests with staggering? And are there any other issues left? Would be nice to get this PR merged :-)